smartsim/_core/control/controller.py

```diff
@@ -591,3 +599,113 @@ def reload_saved_db(self, checkpoint_file):
             return orc
         finally:
             JM_LOCK.release()
+
+
+    def _enumerate_devices(self, dbobject: DBObject):
```

Comment:
I'm thinking this might be better as a method within ``DBObject``.

Reply:
Moved it